parser.go

```diff
@@ -1,29 +1,111 @@
 package main
 
 import (
-	"strings"
-	"log"
+	"encoding/json"
+	"fmt"
 	req "github.com/imroc/req"
+	"io/ioutil"
+	"os"
+	"strings"
 )
 
-func nemoParser(pesan string, Sessions Session) (string, error){
-	for indexURL := 0; indexURL < strings.Count(pesan, "{{"); indexURL++ {
-		url := between(pesan, "{{", "}}")
+var (
+	// TODO use struct for key priority to allow setting config via file
+	// that might need extra config validation though
+	defSupportedRespKeys = map[string]int{
+		"message": 1,
+	}
+	defSupportedRespKeysConfig = "SUPPORTED_KEYS_CONFIG"
+
+	// TODO handle error differently
+	// these are only for visualization purpose
+	errReqErr           = func(dst string) string { return fmt.Sprintf("[req error: %s]", dst) }
+	errRespNotSupported = func(dst string) string { return fmt.Sprintf("[resp not supported error: %s]", dst) }
+)
 
+// nemoParser parses pesan for URL in {{url}} format
+// if no URL is found in pesan, pesan is returned as is
+// if URL is found, try POST request to url
+// currently only supports JSON response with message key
+func nemoParser(pesan string, Sessions Session) (string, error) {
+	urlCount := strings.Count(pesan, "{{")
+	if urlCount == 0 {
+		return pesan, nil
+	}
+	for i := 0; i < urlCount; i++ {
+		url := between(pesan, "{{", "}}")
 		r, err := req.Post(url, req.BodyJSON(Sessions))
 		if err != nil {
-			return "error occured, try again", err 
+			pesan = strings.Replace(pesan, fmt.Sprintf("{{%s}}", url), errReqErr(url), -1)
```

Comment:
if there are multiple URLs and one of the request fails, replace current URL in pesan with error message and continue. change this as you need.